Edits to Closed PR 1414 (backend-api file). (#1816)
vaadin/docs: articles/tools/ce/advanced/backend-api.asciidoc

.Experimental feature
[IMPORTANT]
The Collaboration Engine backend for clustering support is an experimental feature. This means that its behavior, API, as well as its look and feel might change.
You may enable it with a <<Feature Flag>>.

> Using Collaboration Engine with its default settings in an application running in a clustered environment would result in users being able to collaborate only with others connected to the same application instance (i.e., on the same node).

Reviewer: We try to avoid using the latin abbreviations `e.g.` and `i.e.`, as those are not necessarily easy for non-native English readers to understand.

And parenthesis are often unnecessary. I don't mean to be strict about these, but I think it’s always good to evaluate the need for them.

I wonder if there’s a better way of explaining the relationship between an application instance and a server node?

```suggestion
Using Collaboration Engine with its default settings in an application running in a clustered environment would result in users being able to collaborate only with others connected to the same application instance, that is, on the same node.
```

The Event Log can be implemented to make use of a Hazelcast [classname]`IList`. However, first you need a class to store both the event identifier and payload.

[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/ce/hazelcast/HazelcastBackend.java[tags=id-payload,indent=0]
----

Once you have that, you can start implementing the interface methods.
The [methodname]`submitEvent` uses the event identifier and payload, so you can store them in a new [classname]`IdAndPayload` object. You may also add it to the Hazelcast list for this Event Log.

[source,java]
----
include::{root}/src/main/java/com/vaadin/demo/ce/hazelcast/HazelcastBackend.java[tags=event-log;submit-event;!*,indent=0]
----

To implement subscriptions to events, you would add an item listener to the Hazelcast list.
The subscriber will receive all past and future events for this Event Log.
A newly added subscriber should receive initially all of the previous events in the log based on their original order. This allows it to catch up to the latest state.
New events should be delivered, in order, only after all previous events have been delivered.

> The [methodname]`subscribe` method may not be invoked again until the previous subscriber has been removed.

Reviewer: We try to avoid the future tense (will, won’t). Is it actually needed here?

I think “would” and “wouldn’t” fall into similar category. Can those be replaced with “can” and “can’t”?

Author: I think it works better here.
